[wip] vSphere - Use extra_config inplace of vApp; use terraform file for ign

jcpowermac opened the pull request:
NOTE: This will fail CI

- Remove the use of the bootstrap url
- Remove the use of vApp properties, replace with extra_config.
  Limitations with vApp properties max length of 64kb
- In place of ignition files in variables.tf use terraform file
  to read files from `./ignition` directory

abhinavdahiya commented on upi/vsphere/main.tf:
```diff
@@ -29,7 +29,7 @@ module "bootstrap" {
   name             = "bootstrap"
   instance_count   = "${var.bootstrap_complete ? 0 : 1}"
-  ignition_url     = "${var.bootstrap_ignition_url}"
+  ignition         = "${file("${path.module}/ignition/bootstrap.ign")}"
```
hmm, should we be using a variable for this? this makes the ignition in `{root}/ignition/bootstrap.ign` ??

who puts that file there?

abhinavdahiya replied:
we should take a variable the points the file's location..

jcpowermac replied:
What I was thinking was copying the ignition files to a certain location within the terraform directory path to reduce the quantity of variables that a user would need to modify.
Sure I could create a variable to hold the path.  Then in the terraform.tfvars.example use `${path.module}/ignition/node_type.ign`? Or do you not want to modify the directory structure within the `./upi/vsphere` ?

jstuever replied:
Perhaps everything can be done inside of the same (upi/vsphere) directory? (openshift-install create ignition-configs + terraform apply) You could have variables passed into main.yaml with default path ${path.module}/node_type.ign.

abhinavdahiya replied:
personally, i would like users to provide the filepath, and fail if they don't.
Because the installer generates all the ign files in a directory and the terraform {root} might be in a separate one, it makes sense to ask users to give us the location.

I'm fine if we think the user can point to the directory of the install where all the assets are, but explicit paths are more sustainable.

dav1x commented:
I tested this today and it works great. 

LGTM

jcpowermac commented:
I need to retest this.  Removed the restart [0] and added the new vars to the path of the ignition files.
[0] - https://github.com/openshift/installer/pull/2554#issuecomment-545596094

DanyC97 commented:
@jcpowermac you still okay to get this work merged ?

abhinavdahiya commented:
Closing due to this being open for a long time, Please feel free to reopen

/close
